src/array_unique.rs

```diff
@@ -0,0 +1,49 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2022
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+use hashbrown::HashMap;
+use numpy::PyReadonlyArray2;
+
+#[pyfunction]
+#[pyo3(text_signature = "(array)")]
+pub fn unique(array: PyReadonlyArray2<u16>) -> PyResult<(Vec<usize>, Vec<usize>)> {
```

Comment:
You don't need a `PyResult` here unless something can return an error here and you want to return an exception to python. Since it doesn't look there are any error returns it's easier to just drop the `PyResult` and make the return:

```suggestion
pub fn unique(array: PyReadonlyArray2<u16>) -> <(Vec<usize>, Vec<usize>) {
```

Reply:
Thanks. I use this.